osu.Game.Rulesets.Catch/UI/CatcherArea.cs

```diff
@@ -43,31 +44,42 @@ public CatcherArea(BeatmapDifficulty difficulty = null)
             };
         }
 
+        private DrawableCatchHitObject lastPlateableFruit;
+
         public void OnJudgement(DrawableCatchHitObject fruit, Judgement judgement)
         {
-            if (judgement.IsHit)
+            if (judgement.IsHit && fruit.CanBePlated)
             {
-                var screenSpacePosition = fruit.ScreenSpaceDrawQuad.Centre;
+                var caughtFruit = (DrawableCatchHitObject)GetVisualRepresentation?.Invoke(fruit.HitObject);
+
+                if (caughtFruit == null) return;
 
-                // todo: make this less ugly, somehow.
-                (fruit.Parent as HitObjectContainer)?.Remove(fruit);
-                (fruit.Parent as Container)?.Remove(fruit);
+                caughtFruit.State.Value = ArmedState.Idle;

```

Comment:
Is this state setting really required? This is the default state for new hitobjects.

Reply:
don't believe it is, will remove.